src/scss/_checkbox.scss

```diff
@@ -0,0 +1,124 @@
+/// @access private
+/// @param {Boolean} $additional-content Whether to output additional content styling
+/// @param {Boolean} $disabled Whether to output disabled state styling
+/// @param {Boolean} $inline-children Whether to output element inputs inline
+/// @param {Boolean} $inline-label Whether to output entire element inline with its label
+/// @param {Boolean} $invalid Whether to output invalid state styling
+/// @param {Boolean} $right Whether to display checkbox to the right of text
+/// @output Styling for text inputs
+@mixin _oFormsCheckbox(
+	$additional-content: null,
+	$disabled: null,
+	$inline-children: null,
+	$inline-label: null,
+	$invalid: null,
+	$right: null
+) {
+	.o-forms--checkbox {
+		label span {
+			display: inline-block;
+			max-width: calc(100% - 34px);
+			padding: 2px 0 2px 34px;
+			vertical-align: top;
+
+			&:before,
+			&:after {
+				@include _oFormsControlsPseudoElements();
+			}
+
+			&:before {
+				border: 1px solid _oFormsGet('default-border');
+				background-color: _oFormsGet('default-background');
+			}
+
+			&:after {
+				@include oIconsGetIcon('tick', _oFormsGet('controls-checked-base'), 24px);
+				background-color: _oFormsGet('controls-base');
+				opacity: 0;
+				transition: opacity 0.1s ease;
+			}
+		}
+
+		@include oGridRespondTo(S) {
+			label:first-of-type {
+				margin: 12px 0;
+			}
+		};
+
+		input[type=checkbox] { // sass-lint:disable-line no-qualifying-elements
+			@include _oFormsControlsBase($disabled);
+			float: left;
+		}
+
+		@if $additional-content {
+			&.o-forms--checkbox--extra-content {
+				> *:not(legend) {
+					margin-bottom: 10px;
+				}
+
+				input[type=checkbox] { // sass-lint:disable-line no-qualifying-elements
+					display: inline-block;
+					vertical-align: top;
+				}
+
+				p {
+					margin-top: 10px;
+					padding-top: 2px;
+					padding-left: 34px;
+				}
+			}
+		}
+
+		@if $inline-children {
+			&.o-forms--inline-children {
+				@include _oFormsInlineChildren();
+			}
+		}
+
+		@if $inline-label {
+			&.o-forms--inline {
+				@include _oFormsInlineLabel($fieldset: true);
+			}
+		}
+
+		@if $invalid {
+			&.o-forms--invalid .o-forms__error-message {
+				@include oTypographyMargin($top: 2);
+			}
+		}
+
+		@if $right {
+			&.o-forms--checkbox--right {
```

Comment:
Should this be `o-forms--checkbox-right`? It's a single BEM modifier that's _kind_ of masquerading as two. Can a modifier have a modifier of its own? Just the first instance I found but saw a few more in the demos, worth a discussion maybe